meta-lmp-base/recipes-core/initrdscripts/initramfs-framework/ostree_recovery

```diff
@@ -7,8 +7,44 @@ ostree_recovery_enabled() {
 }
 
 ostree_recovery_run() {
-	msg "Starting ostree recovery shell..."
-	sh
+	RECOVERY_MODULES_DIR=/recovery.d
+
+	# Load and run recovery modules same way as done for initramfs
+	for m in `ls $RECOVERY_MODULES_DIR/ | sort -n`; do
+		# Skip backup files
+		if [ "`echo $m | sed -e 's/\~$//'`" != "$m" ]; then
+			continue
+		fi
+
+		module=`basename $m | cut -d'-' -f 2`
+		debug "Loading recovery module $module"
+
+		# pre hooks
+		for h in $MODULE_PRE_HOOKS; do
```

Comment:
`MODULE_PRE_HOOKS` and `MODULE_POST_HOOKS` are in initramfs-framework init but we don't use them.
I am think in decouple the https://cgit.openembedded.org/openembedded-core/tree/meta/recipes-core/initrdscripts/initramfs-framework/init in order to enable the use some of its functions.

```
MODULE_PRE_HOOKS=""  # functions to call before running each module
MODULE_POST_HOOKS="" # functions to call after running each module
```

Reply:
We actually use it because ostree_recovery is an initramfs-framework module as well, and why I extended this logic to make the same modules and hooks available for the recovery.d logic.

Reply:
got it, thanks for the explanation